perimeter/public_access.sp

```diff
@@ -461,102 +461,62 @@ control "s3_bucket_acl_prohibit_public_write_access" {
-        )
-        -- aws:PrincipalOrgID
-        and s -> 'Condition' -> 'StringEquals' -> 'aws:principalorgid' is null
-        and s -> 'Condition' -> 'StringEqualsIgnoreCase' -> 'aws:principalorgid' is null
-        and (
-          s -> 'Condition' -> 'StringLike' -> 'aws:principalorgid' is null
-          or s -> 'Condition' -> 'StringLike' -> 'aws:principalorgid' ? '*'
-        )
-        -- aws:PrincipalAccount
-        and s -> 'Condition' -> 'StringEquals' -> 'aws:principalaccount' is null
-        and s -> 'Condition' -> 'StringEqualsIgnoreCase' -> 'aws:principalaccount' is null
-        and (
-          s -> 'Condition' -> 'StringLike' -> 'aws:principalaccount' is null
-          or s -> 'Condition' -> 'StringLike' -> 'aws:principalaccount' ? '*'
-        )
-        -- aws:PrincipalArn
-        and s -> 'Condition' -> 'StringEquals' -> 'aws:principalarn' is null
-        and s -> 'Condition' -> 'StringEqualsIgnoreCase' -> 'aws:principalarn' is null
-        and (
-          s -> 'Condition' -> 'StringLike' -> 'aws:principalarn' is null
-          or s -> 'Condition' -> 'StringLike' -> 'aws:principalarn' ? '*'
-        )
-        and (
-          s -> 'Condition' -> 'ArnEquals' -> 'aws:principalarn' is null
-          or s -> 'Condition' -> 'ArnEquals' -> 'aws:principalarn' ? '*'
-        )
-        and (
-          s -> 'Condition' -> 'ArnLike' -> 'aws:principalarn' is null
-          or s -> 'Condition' -> 'ArnLike' -> 'aws:principalarn' ? '*'
-        )
-        -- aws:SourceArn
-        and s -> 'Condition' -> 'StringEquals' -> 'aws:sourcearn' is null
-        and s -> 'Condition' -> 'StringEqualsIgnoreCase' -> 'aws:sourcearn' is null
-        and (
-          s -> 'Condition' -> 'StringLike' -> 'aws:sourcearn' is null
-          or s -> 'Condition' -> 'StringLike' -> 'aws:sourcearn' ? '*'
-        )
-        and (
-          s -> 'Condition' -> 'ArnEquals' -> 'aws:sourcearn' is null
-          or s -> 'Condition' -> 'ArnEquals' -> 'aws:sourcearn' ? '*'
-        )
-        and (
-          s -> 'Condition' -> 'ArnLike' -> 'aws:sourcearn' is null
-          or s -> 'Condition' -> 'ArnLike' -> 'aws:sourcearn' ? '*'
-        )
-        and (
-          s -> 'Principal' -> 'AWS' = '["*"]'
-          or s ->> 'Principal' = '*'
-        )
-      group by
-        __ARN_COLUMN__
-    )
+  resource_policy_public_sql = <<-EOT
     select
       r.__ARN_COLUMN__ as resource,
       case
-        when p.__ARN_COLUMN__ is null then 'ok'
-        else 'alarm'
+        when pa.is_public = true then 'alarm'
+        else 'ok'
       end as status,
       case
-        when p.__ARN_COLUMN__ is null then title || ' policy does not allow public access.'
-        else title || ' policy contains ' || coalesce(p.statements_num, 0) ||
-        ' statement(s) that allow public access.'
+        when pa.is_public = false then title || ' policy does not allow public access.'
+        when jsonb_array_length(pa.public_statement_ids) = 1 then concat(
```

Comment:
If we changed all of the `alarm` cases to use the following format:
```sql
else title || ' policy has ' || jsonb_array_length(pa.public_statement_ids) || ' statements that allow public access: [' || pa.public_statement_ids || '].'
```

So some sample reasons would look like:
- `s3-bucket-test policy has 1 statement(s) that allow public access: [MySID1].`
- `s3-bucket-test policy has 3 statement(s) that allow public access: [MySID1, Statement[2], SampleSID].`
- `s3-bucket-test policy has 10 statement(s) that allow public access: [MySID1, Statement[2], PublicAccess, Testing, Foo, Bar, F1, F2, F3, EndingSID].`

Does the console + dashboard Ui output render this without any errors? I'm expecting it to get cut off if the list of items is longer than 2 or 3, but I think this may be ok since our current set of reasons only show this many anyway.

Also, I believe users can output to CSV, JSON, or another format, in order to get the full reason (need to verify this though).